presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/AlternativeNullableLongStateSerializer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.function.AccumulatorStateSerializer;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+
+public class AlternativeNullableLongStateSerializer
```

Comment:
It looks like the big difference between this and NullableLongStateSerializer is that this one uses the block.toLong() method, and the other uses the block.getLong() method?  Would be helpful to have some Java doc at the top of the class explaining the purpose of the class, how it's different from NullableLongStateSerializer and why.